### DISCUSSION TITLE
Running Inference Stopped Working: `Unable to load frame`

### DISCUSSION THREAD

**Discussion by wallawhitecoat** [talmolab/sleap#1944] [General]
Hello, I was running my SLEAP model, about the 5th time doing so on a series of videos. I ran into this issue, and now all of the videos I plug in to run inference are showing me the same error message. I'm not sure what's wrong as I haven't changed anything, and it was running smoothly earlier today. This error only appears once the model is predicting. 
![image](https://github.com/user-attachments/assets/0311237c-b4c1-476c-a5e9-ce95c44026b7)
Any help would be greatly appreciated, thank you!

Windows 10, version 22H2
Python Version [3.10.14]
using miniconda3 Version [24.4.0]

**Comment by eberrigan** [comment]
Hi @wallawhitecoat,

How did you install SLEAP?

Thanks!

Elizabeth

**Comment by roomrys** [comment]
Hi @wallawhitecoat,

I see you are getting a message similar to:
```
KeyError: "Unable to load frame X from MediaVideo"
```
.

We have seen a few problems like this occur when the video is not [reliably seekable](http://blog.loopbio.com/video-io-1-introduction.html). Try [reencoding your video](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) and let us know how that works.

---
#### Update: If reencoding doesn't work, and results in `moov atom not found`
As I was helping a user, we found that they had just copied the contents of a zipped file over without unzipping which resulted in first the `KeyError`, and then, when attempting to reencode, a `moov atom not found` error. After retrying but with **unzipping first**, we were able to get everything running smoothly.

---

Thanks,
Liezl

P.S. Also linking similar problems here for reference (namely KeyError "Unable to load frame" causing SLEAP to crash when predicting
#767, Unable to load frame #366, KeyError when trying to run inference #630, Encoding Multiple Videos using Batch File #719, Prediction all frames in a video and processed to 99% and error #742, Unable to load second to last frame of entire current video
#1895).

**Comment by diego-mediane** [comment]
Hello, I'm experiencing the same issue and I've noticed that the error typically occurs with the second-to-last frame. A solution I've found is to "trim" the final few frames from the video.

If you're using Windows 11, you can open the video in the "Photos" app, and you'll find the trim option in the top left corner. For Mac users, the default media player has a built-in trim feature under "edit", but remember to save the trimmed video as a new file.
